### PR TITLE
implement python injection test matrix [APMON-1544]

### PR DESCRIPTION
**Pull request by emmettbutler**
This change adds configuration combinations to the installer test matrix for Python, testing all supported Python runtime versions, amd64/arm64, and glibc/musl.

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/_context/virtual_machines.py [hunk 1]
```diff
@@ -553,7 +553,7 @@ def __init__(self, **kwargs) -> None:
             krunvm_config=None,
             os_type="linux",
             os_distro="deb",
-            os_branch="ubuntu24",
+            os_branch="ubuntu24_amd64",
```
Comment (robertomonteromiguel) [inline utils/_context/virtual_machines.py:556]: why this change?

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]

File: utils/build/virtual_machine/weblogs/python/test-app-python-container-3.10-alpine/Dockerfile.template [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM public.ecr.aws/docker/library/python:3.10
```
Comment (robertomonteromiguel) [inline utils/build/virtual_machine/weblogs/python/test-app-python-container-3.10-alpine/Dockerfile.template:1]: should be python:3.10-alpine???

File: utils/build/virtual_machine/weblogs/python/test-app-python-container-3.11-alpine/Dockerfile.template [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM public.ecr.aws/docker/library/python:3.11
```
Comment (robertomonteromiguel) [inline utils/build/virtual_machine/weblogs/python/test-app-python-container-3.11-alpine/Dockerfile.template:1]: should be xxxx-alpine?

File: utils/build/virtual_machine/weblogs/python/provision_test-app-python-container-3.10.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+weblog:
+    name: test-app-python-container-3.10
+    exact_os_branches: [ubuntu24]
```
Comment (robertomonteromiguel) [inline utils/build/virtual_machine/weblogs/python/provision_test-app-python-container-3.10.yml:3]: I think we should try to run on multiple machines

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]

File: utils/build/virtual_machine/weblogs/python/provision_test-app-python-container-3.10.yml (hunk 4 above)
Comment (emmettbutler) [reply utils/build/virtual_machine/weblogs/python/provision_test-app-python-container-3.10.yml:3]: For now I'm following the guidance in the linked JIRA ticket, which is to only test these against latest ubuntu.
